### DISCUSSION TITLE
Can we speed up `vscode.workspace.findFiles`?

### DISCUSSION THREAD

**Discussion by bilogic** [microsoft/vscode-discussions#1179] [Extension Development QnA] [answered]
Is there a way to cache and add onto `vscode.workspace.findFiles` instead of having to re-run it each time?

**Comment by isidorn** [comment, marked as answer]
This is currently not possible. You can build a cache on your extension layer.
What's your use case? How come findFiles is slow?

**Reply by bilogic** [reply]
Use case: I'm looking for files that match a certain pattern

I have since mirrored a full list of files in the memory that also self updates based on file watching, that took the search from 10-30 secs down til immediate. In my view, this would be helpful if implemented in `findFiles`, because I'm not confident my file watching is bug free.
